extensions/cdc-debezium/src/main/java/com/hazelcast/jet/cdc/impl/ChangeRecordCdcSourceP.java

```diff
@@ -59,34 +53,37 @@ public ChangeRecordCdcSourceP(
     @Nullable
     @Override
     protected ChangeRecord map(SourceRecord record) {
-        record = transform.apply(record);
         if (record == null) {
             return null;
         }
 
         long sequenceSource = sequenceExtractor.source(record.sourcePartition(), record.sourceOffset());
         long sequenceValue = sequenceExtractor.sequence(record.sourceOffset());
         String keyJson = Values.convertToString(record.keySchema(), record.key());
-        String valueJson = Values.convertToString(record.valueSchema(), record.value());
-        return new ChangeRecordImpl(sequenceSource, sequenceValue, keyJson, valueJson);
-    }
-
-    private static ExtractNewRecordState<SourceRecord> initTransform(String dbSpecificExtraFields) {
-        ExtractNewRecordState<SourceRecord> transform = new ExtractNewRecordState<>();
+        Struct value = (Struct) record.value();
+        Struct source = (Struct) value.get("source");
 
-        Map<String, String> config = new HashMap<>();
-        config.put("add.fields", String.join(",", extraFields(dbSpecificExtraFields)));
-        config.put("delete.handling.mode", "rewrite");
-        transform.configure(config);
-
-        return transform;
+        Operation operation = Operation.get(value.getString("op"));
+        Schema valueSchema = record.valueSchema();
+        String oldValueJson = Values.convertToString(valueSchema.field("before").schema(), value.get("before"));
+        String newValueJson = Values.convertToString(valueSchema.field("after").schema(), value.get("after"));
```

Comment:
We should try to reduce the amount of garbage created - not all will use both before & after, not all fields will be read,
I wonder if this can be improved, e.g. by converting lazily, or not converting at all if not needed.

Reply:
I may save the json value and convert lazily using some generic JSON mapper (jackson/gson). IIRC schema is not even requried, so this should not be a problem